src/java.desktop/share/classes/javax/swing/plaf/synth/SynthLookAndFeel.java

```diff
@@ -728,6 +728,8 @@ public UIDefaults getDefaults() {
                              SwingUtilities2.getSystemMnemonicKeyMask())
                   });
 
+        table.put("PasswordField.echoChar", '*');
```

Comment:
It is still unclear where this "*" was stored before(will we ignore it now?), since the Nimbus used that symbol already.

Reply:
"*" is the default char unless overridden by echoChar property and is set [here](https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/javax/swing/JPasswordField.java#L178)

Reply:
I just found my comment at that bug that the ui properties for nimbus are stored in the "skin.laf" and the "echoChar" is missed there. This is the reason why the UIDefaults table is small in Nimbus, unlike Metal where we have a long list of properties. Take a look to the `NimbusDefaults.java` which is generated during the build it contains all the props, and it is a place where I think we should update the echoChar.

Reply:
It will have the problem as you mentioned so I did it as it is. Since we have already UIDefaults table defined, I think it can be used for this, without changing the doc which might cause compatibility issues.

Reply:
it will also probably need change in compiler as to how it generates/parses the type as I am getting

Exception in thread "main" java.lang.NullPointerException: Cannot invoke "build.tools.generatenimbus.UIProperty$PropertyType.ordinal()" because "this.type" is null
	at build.tools.generatenimbus.UIProperty.write(UIProperty.java:87)
	at build.tools.generatenimbus.UIStyle.write(UIStyle.java:156)
	at build.tools.generatenimbus.UIRegion.write(UIRegion.java:213)
	at build.tools.generatenimbus.SynthModel.write(SynthModel.java:102)
	at build.tools.generatenimbus.Generator.generate(Generator.java:212)
	at build.tools.generatenimbus.Generator.main(Generator.java:118)

so I think this is the safer and cleaner alternative as I am not creating a new UIDefaults table, just adding a new entry to it..

Reply:
It is probably simpler but we already store all tha properties in the generated NimbusDefaults, I do not think we should split the storage in two parts.

Reply:
Properties are **already** being stored in two parts so I am not inventing it for this fix. I am just adding an entry to the already created UIDefaults. Also, as I told, storing a character in skin.laf involves changing the generator tool which might not be scalable and prone to regressions, not to mention what/where should be changed in the tool for this is not known to me.

Also, as Synth is a skinnable look and feel, I think it would have been an issue if the property to be added affects "skin" in anyways, like **color, font, dimension** property etc, in which case it would have made sense to be mandatorily put in skin.laf which is the "visual designer"
 but this Password echo character property will not affect the skin in anyway, like the other properties added in UIDefaults table in SynthLookAndFeel, so I think it should be ok.

Reply:
I agree with @prsadhuk here. The `echoChar` isn't skin, it can be defined directly as is done for other properties.

Reply:
Most of that "other" properties are specified by the "skin" file for the Nimbus L&F. Take a look to the "NimbusDefaults.java", it has many-many properties which are usually hardcoded in other L&F. Inital idea of the "skin" was the ability to swicth the UI appiarance, the style of "echo" is part of it.

It is of course possible to set/update that field directly by the application, but for the Nimbus L&F default value should be set by the skin.